app/src/main/java/com/arctouch/publictransportation/fragment/StopListFragment.java

```diff
@@ -0,0 +1,185 @@
+package com.arctouch.publictransportation.fragment;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.android.volley.RequestQueue;
+import com.arctouch.publictransportation.R;
+import com.arctouch.publictransportation.activity.DepartureListActivity;
+import com.arctouch.publictransportation.domain.Route;
+import com.arctouch.publictransportation.domain.Stop;
+import com.arctouch.publictransportation.domain.generic.params.RouteIdParam;
+import com.arctouch.publictransportation.domain.generic.returns.StopsReturn;
+import com.arctouch.publictransportation.interfaces.VolleyListener;
+import com.arctouch.publictransportation.volley.VolleySingleton;
+import com.arctouch.publictransportation.volley.endpoints.StopsVolley;
+
+import java.util.List;
+
+public class StopListFragment extends Fragment implements VolleyListener<StopsReturn> {
+    public static final String ROUTE = "route";
+    public static final String SHOW_ROUTE_DETAILS = "show_route_details";
+
+    private View mRootView;
+    private View mRecyclerView;
+    private ProgressBar mProgressBar;
+    private TextView mEmptyList;
+    private Button mSeeDepartures;
+
+    private boolean mShowRouteDetails;
+
+    private Route mRoute;
+
+    public StopListFragment() {
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (getArguments().containsKey(ROUTE)) {
```

Comment:
Same as in DepartureListFragment - This check would become unnecessary with a Factory Method